core/templates/pages/topic-editor-page/questions-tab/topic-questions-tab.component.spec.ts

```diff
@@ -0,0 +1,224 @@
+    spyOn(focusManagerService, 'setFocus');
+    spyOnProperty(TopicEditorStateService, 'onTopicInitialized').and.callFake(
+      function() {
+        return topicInitializedEventEmitter;
+      });
+    spyOnProperty(
+      TopicEditorStateService, 'onTopicReinitialized').and.callFake(
+      function() {
+        return topicReinitializedEventEmitter;
+      });
+    ctrl.$onInit();
+  }));
+
+  afterEach(() => {
+    ctrl.$onDestroy();
+  });
+
+  it('should initialize the variables', function() {
+    const topicRights = TopicRights.createInterstitialRights();
+    const allSkillSummaries = subtopic1.getSkillSummaries();
+    spyOn(TopicEditorStateService, 'getSkillIdToRubricsObject').and
+      .returnValue(skillIdToRubricsObject);
+
+    expect(focusManagerService.setFocus).toHaveBeenCalledWith(
+      'selectSkillField');
+    expect($scope.selectedSkillId).toBeNull();
+    expect($scope.question).toBeNull();
+    expect($scope.skillId).toBeNull();
+    expect($scope.topic).toBe(topic);
+    expect($scope.topicRights).toEqual(topicRights);
+    expect($scope.skillIdToRubricsObject).toEqual(skillIdToRubricsObject);
+    expect($scope.allSkillSummaries).toEqual(allSkillSummaries);
+    expect($scope.getSkillsCategorizedByTopics).toBe(CategorizedSkillsDict);
+    expect($scope.getUntriagedSkillSummaries)
+      .toBe(UntriagedSkillSummaries);
+    expect($scope.canEditQuestion).toBe(false);
+    expect($scope.misconceptions).toEqual([]);
+    expect($scope.questionIsBeingUpdated).toBe(false);
+    expect($scope.questionIsBeingSaved).toBe(false);
+    expect($scope.emptyMisconceptionsList).toEqual([]);
+  });
+
+  it('should setFocus on selectSkillField when screen loads', function() {
+    $window.onload();
+
+    expect(focusManagerService.setFocus).toHaveBeenCalledWith(
+      'selectSkillField');
+  });
+
+  it('should call initTab when topic is initialized', function() {
+    $scope.question = 'question1';
+    $scope.skillId = '1';
+    $scope.topic = null;
+
+    topicInitializedEventEmitter.emit();
+
+    expect($scope.question).toBeNull();
+    expect($scope.skillId).toBeNull();
+    expect($scope.topic).toBe(topic);
+  });
+
+  it('should call initTab when topic is reinitialized', function() {
+    $scope.question = 'question1';
+    $scope.skillId = '1';
+    $scope.topic = null;
+
+    topicInitializedEventEmitter.emit();
+    expect($scope.question).toBeNull();
+    expect($scope.skillId).toBeNull();
+    expect($scope.topic).toBe(topic);
+    $scope.question = 'question1';
+    $scope.skillId = '1';
+    $scope.topic = null;
+    topicReinitializedEventEmitter.emit();
+
+    expect($scope.question).toBeNull();
+    expect($scope.skillId).toBeNull();
+    expect($scope.topic).toBe(topic);
+  });
+
+  it('should unsubscribe when onDestroy runs', function() {
+    spyOn(ctrl.directiveSubscriptions, 'unsubscribe');
+
+    ctrl.$onDestroy();
+
+    expect(ctrl.directiveSubscriptions.unsubscribe).toHaveBeenCalled();
```

Comment:
Better if you check that the list of directive subscriptions was nonempty before the call and empty after.

In general, prefer checking the actual states of things and not just that some method has been called, where possible.

Reply:
I'm not sure how to test that. Any tips?

Reply:
If you're not sure how to do something, always create a [debugging doc](https://github.com/oppia/oppia/wiki/Debugging-Docs) explaining what you tried and what you are thinking, and then I (or someone else) can suggest next steps. With this, I have too little to go on and can't really help you.

My first inclination would have been to see if I can introspect ctrl.directiveSubscriptions to see if I can get a list of subscriptions that are still active; this requires you to look into the API of the Subscription object etc.

Reply:
DONE